R/with_parameters.R

```diff
@@ -155,9 +170,21 @@ build_label <- function(..., case_names) {
   toString(sprintf("%s=%s", case_names, row))
 }
 
-build_and_run_test <- function(..., .test_name, desc, code, env) {
+build_and_run_test <- function(..., .test_name, desc, code, env, .interpret_glue) {
   args <- list(..., .test_name = .test_name)
-  completed_desc <- glue_data(args, desc)
+  if (.interpret_glue) {
+    completed_desc <- tryCatch(glue_data(args, desc), error = identity)
+    if (inherits(completed_desc, "error")) {
+      rlang::abort(sprintf(
+        paste(
+          "Attempt to interpret test stub '%s' with glue failed with error:\n%s\n\n",
+          "Set .interpret_glue=FALSE if this test name does not use glue."
+        ),
+        # indent for clarity (the purrr error has similar mark-up)
+        desc, gsub("(^|\n)", "\\1  ", conditionMessage(completed_desc))
+      ))
+    }
+  }
```

Comment:
@MichaelChirico Thanks! Would you mind making the other logical branch explicit. Right now I cant really tell how completed_desc is getting set, and I think something funny is happening at line 189 to get the test case to pass.

Reply:
Oh, great catch... possibly it's a {testthat} thing about how `expect_success()` works...

Reply:
Line 198 has one more call glue that needs to be caught when .interpret_glue = FALSE.

@MichaelChirico I'm glad the tests were able to pick that up.

Reply:
good call. actually, build_description is pretty cluttered with this logic for constructing the description at this point. went ahead and ported that to a helper instead.

also, I think the below TODO about testthat_braces_warning is long superseded -- that PR first hit {testthat} 4 years ago:

https://github.com/r-lib/testthat/commit/74a0fee43ba1d988ea6b5a03cdb0fbf4824d5dbd

Incremented the testthat required version to be sure.